### PR TITLE
Change light brightness, control light groups, create user

### PR DESCRIPTION
**Pull request by roguh**
- Add ability to change light brightness
- Add http.post and bridge.create_user
- Add ability to query and control light groups

### CODE REVIEW

**Comment by roguh** [comment]
Hello, would you like to accept these changes? I can also split them up into multiple PRs. Any feedback is appreciated. I can make any changes you need.

**Comment by roguh** [comment]
There's almost certainly a more efficient way of toggling on all lights a group. I can research this more before you consider this PR, if you want.

**Comment by nirantak** [comment]
> Hello, would you like to accept these changes? I can also split them up into multiple PRs. Any feedback is appreciated. I can make any changes you need.

Hi @roguh, I will be happy to accept any changes to this project. It might take a couple of days to review this PR, but in the meanwhile can you please add tests for these changes and fix the Linter check?

**Comment by nirantak** [comment]
> There's almost certainly a more efficient way of toggling on all lights a group. I can research this more before you consider this PR, if you want.

It would be great if you could do this

**Comment by roguh** [comment]
I'll add tests, run the linter locally, and find a better way to control a light group.

**Comment by roguh** [comment]
I've added tests and fixed the light group control so it is far more efficient and reliable.

I still need to add tests to the light group code.

The only code of yours that I've changed is where you sent `"on"` in every call to set_state. I think it's better to only send `"on"` when turning a light on or off.

File: hue/api/light.py [hunk 1]
```diff
@@ -153,3 +157,13 @@ async def toggle(self) -> list[dict[str, dict[str, Any]]]:
...
         return await self.set_state({"on": not self.on})
+
+    async def set_brightness(self, brightness: int) -> list[dict[str, dict[str, Any]]]:
+        """
+        Power on the Light
+
+        Returns:
+            A list of dictionaries with key=success/error and value=state element changed
+        """
+        await self.get_state()
```
Comment (nirantak) [inline hue/api/light.py:168]:
```suggestion
```
This line is not needed

Comment (nirantak) [reply hue/api/light.py:168]: Also, please do update the docstring for description and arguments

File: hue/api/light_group.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+class LightGroup(Bridge):
+    """
+    Interact with the Hue Lights API
+
+    Example:
+        >>> from hue import LightGroup
+        >>> bridge = LightGroup("All", ip="192.168.1.10", user="xxxx")
+
+    Attributes:
+        name: Name of the Hue Light group
```
Comment (nirantak) [inline hue/api/light_group.py:18]:
```suggestion
    Interact with the Hue Light Groups API

    Example:
        >>> from hue import LightGroup
        >>> group = LightGroup("All", ip="192.168.1.10", user="xxxx")

    Attributes:
        name: Name of the Light Group
```

File: hue/api/light_group.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        """
+        Initialize a Hue Light object
+
+        Arguments:
+            name: The name of the Light group to control
+            ip: IP address of the Hue Bridge, get it by running `hue bridge discover`
+            user: The secret user id created to access the Hue API
+
+        Returns:
+            An object of class Light
```
Comment (nirantak) [inline hue/api/light_group.py:33]:
```suggestion
        Initialize a Hue Light Group object

        Arguments:
            name: The name of the Light Group to control
            ip: IP address of the Hue Bridge, get it by running `hue bridge discover`
            user: The secret user id created to access the Hue API

        Returns:
            An object of class LightGroup
```

File: hue/api/light_group.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    @staticmethod
+    async def list(ip: str, user: str):
+        groups = (await http.get(f"http://{ip}/api/{user}/groups")).json()
+        try:
+            return [group["name"] for group in groups.values()]
+        except (AttributeError, KeyError) as exc:
+            print(exc)
+        return None
+
+    async def set_state(self, state):
```
Comment (nirantak) [inline hue/api/light_group.py:60]: Can you please add type hints for all method arguments and return values?

File: hue/cli/bridge.py [hunk 5]
```diff
@@ -36,3 +36,16 @@ def get(
...
+
+
+@app.command()
+def create_user(
+    ip: str = typer.Option(..., "--ip", "-i", envvar="HUE_BRIDGE_IP"),
+    device_type: str = typer.Option(..., "--device-type", "-d"),
+):
+    """Create a user on a Hue Bridge.
+
+    You must press the button on the Bridge.
```
Comment (nirantak) [inline hue/cli/bridge.py:48]:
```suggestion
    """
    Create a user for the Hue Bridge.
    You must press the button on the Bridge atmost 30 seconds before running this command.
```

File: hue/cli/bridge.py [hunk 6]
```diff
@@ -36,3 +36,16 @@ def get(
...
+    """
+    resp = asyncio.run(Bridge.create_user(ip, device_type))
+    console.print(f"[{ip}] Bridge Config:\n{json.dumps(resp, indent=2)}")
```
Comment (nirantak) [inline hue/cli/bridge.py:51]:
```suggestion
    console.print(f"[{ip}] User Config:\n{json.dumps(resp, indent=2)}")
```

File: hue/cli/lights.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
+import asyncio
```
Comment (nirantak) [inline hue/cli/lights.py:1]: Please rename this file to light_group.py to keep the same pattern as the API directory

**Comment by nirantak** [comment]
Hi @roguh, let me know if you plan to work on this, or I can update and start adding these in bits and pieces.

**Comment by roguh** [comment]
My apologies, I haven't found time for this. You can continue this work, if you wish.

**Comment by podulator** [comment]
Hey, any chance of this getting merged at all, it would be super useful for a pet project :)

**Pull request closed without merging**